interview-mr/src/main/java/com/sovrn/interview/mr/AverageScore.java

```diff
@@ -47,22 +51,49 @@ public static void main(String[] args) throws Exception {
 
     public static class AverageScoreMapper extends Mapper<Object, Text, Text, FloatWritable> {
 
+        private final FloatWritable number = new FloatWritable();
+        private final Text domainText = new Text();
+
         @Override
         protected void map(Object key, Text value, Mapper<Object, Text, Text, FloatWritable>.Context context)
                 throws IOException, InterruptedException {
-            // TODO : Implement the mapping phase
-            
+            String[] values = value.toString().split("\t");
+
+            try {
+                String domain = new URI(values[3]).getHost();
+                number.set(Float.parseFloat(values[1]));
+
+                if(domain != null)
+                    domainText.set(domain);
+                    context.write(domainText, number);
+
+            } catch (URISyntaxException | NumberFormatException e) {
+                // Invalid data - either not a clean URL or the value
+                // wasn't a number (as in the column header line). No
+                // error reporting requirement, so do nothing for now.
+            }
         }
 
     }
 
     public static class AverageScoreReducer extends Reducer<Text, FloatWritable, Text, FloatWritable> {
 
+        private final FloatWritable average = new FloatWritable();
+
         @Override
         protected void reduce(Text key, Iterable<FloatWritable> values,
                 Reducer<Text, FloatWritable, Text, FloatWritable>.Context context) throws IOException,
                 InterruptedException {
-            // TODO: Implemnet the reducer phase
+            float sum = 0f;
+            int count = 0;
+
+            for (FloatWritable value : values) {
+                sum += value.get();
+                count++;
+            }
+
+            average.set(count > 0 ? sum / count : 0f);
```

Comment:
I don't think `reduce` would be called if there were zero values, but divide-by-zero is a nasty condition, so I check for it anyway.